src/librustc_resolve/lib.rs

```diff
@@ -2183,19 +2200,25 @@ impl<'a, 'crateloader: 'a> Resolver<'a, 'crateloader> {
                                              |this| visit::walk_item(this, item));
             }
 
+            ItemKind::Struct(ref variant, ref generics) => {
+                if variant.is_tuple() || variant.is_unit() {
+                    if let Some(def_id) = self.definitions.opt_local_def_id(item.id) {
+                        if let Some(variant_id) = self.definitions.opt_local_def_id(variant.id()) {
+                            let variant_def = if variant.is_tuple() {
+                                Def::StructCtor(variant_id, CtorKind::Fn)
+                            } else {
+                                Def::StructCtor(variant_id, CtorKind::Const)
+                            };
+                            self.tuple_structs.insert(def_id, variant_def);
```

Comment:
Constructors are already to this table in `build_reduced_graph.rs`, so this seems to be duplicate work.

Reply:
Removed.